db/migrate/20170109221226_create_configuration_sources.rb

```diff
@@ -0,0 +1,17 @@
+class CreateConfigurationProjects < ActiveRecord::Migration[5.0]
+  def change
+    create_table :configuration_sources do |t|
+      t.belongs_to  :configuration_manager, :type => :bigint
+      t.string      :manager_ref
+      t.string      :name
+      t.string      :description
+      t.string      :scm_type
+      t.bigint      :scm_reference_id
+      t.boolean     :scm_clean
+      t.boolean     :scm_delete_on_update
+      t.boolean     :scm_update_on_launch
```

Comment:
All of my and @bdunne's comments were lost.

- Why do you have all of these scm_* columns?  None of them should be necessary.
- :scm_reference_id cannot end in `_id` as that is reserved for foreign keys.  Prefer possibly `:scm_ref` (if this column is needed at all)